pkg/webui/console/views/device-general-settings/index.js

```diff
@@ -0,0 +1,106 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import { connect } from 'react-redux'
+import { Col, Row, Container } from 'react-grid-system'
+import bind from 'autobind-decorator'
+import { defineMessages } from 'react-intl'
+
+import sharedMessages from '../../../lib/shared-messages'
+import diff from '../../../lib/diff'
+import api from '../../api'
+
+import DeviceDataForm from '../../containers/device-data-form'
+import IntlHelmet from '../../../lib/components/intl-helmet'
+import toast from '../../../components/toast'
+
+const m = defineMessages({
+  updateSuccess: 'Successfully updated end device',
+})
+
+@connect(function ({ device }, props) {
+  return {
+    device: device.device,
+  }
+})
+@bind
+export default class DeviceGeneralSettings extends React.Component {
+
+  state = {
+    error: undefined,
+  }
+
+  async handleSubmit (values, { setSubmitting, resetForm }) {
+    const { device } = this.props
+    const { application_id } = device.ids.application_ids
```

Comment:
you can selector here, see https://github.com/TheThingsNetwork/lorawan-stack/blob/master/pkg/webui/lib/selectors/id.js#L17-L20